File: Frontend/src/app/relatorios/log-de-processos/models/AgendamentoLogProcesso.model.ts
import { OperacoesEnum } from "../enuns/operacoes.enum";
import { StatusEnum } from "../enuns/status.enum";
import { TipoLogEnum } from "../enuns/tipoLog.enum";

export class AgendamentoLogProcessoModel {

    public id   : number = 0;
    public dataIni : string = "";
    public dataFim : string = "";
    public dataDeAgendamento : string = "";
    public dataDeCadastro : string = "";
    public usuarioId : string = "";
    public operacao : string =  OperacoesEnum["T"];
    public tipoLog : string = TipoLogEnum["0"];
    public status :string = "";
    public nomeDoArquivo : string = "";
    public mensagemErro : string = ""
    public dataIniExecucao : string = "";
    public dataFimExecucao :string = "";


    public static fromObj(obj:any){
        let agendamento = new AgendamentoLogProcessoModel();
        agendamento.id   = obj.id ;
        agendamento.dataIni =  obj.dataIni ,
        agendamento.dataFim =  obj.dataFim ;
        agendamento.dataDeAgendamento =  obj.dataDeAgendamento ;
        agendamento.dataDeCadastro =  obj.dataDeCadastro ;
        agendamento.usuarioId =  obj.usuarioId ;
        agendamento.operacao = Object.keys(OperacoesEnum)[Object.values(OperacoesEnum).indexOf(obj.operacao as unknown as OperacoesEnum)];
        agendamento.tipoLog = Object.keys(TipoLogEnum)[Object.values(TipoLogEnum).indexOf(obj.tipoLog.toString() as unknown as TipoLogEnum)];
        agendamento.status = Object.keys(StatusEnum)[Object.values(StatusEnum).indexOf(obj.status as unknown as StatusEnum)] ;
        agendamento.nomeDoArquivo =  obj.nomeDoArquivo ;
        agendamento.mensagemErro = obj.mensagemErro ;
        agendamento.dataIniExecucao =  obj.dataIniExecucao ;
        agendamento.dataFimExecucao=  obj.dataFimExecucao ;
        return agendamento;
    }

}

File: Backend/V2/Oi.Juridico.WebApi.V2/Areas/Relatorios/Models/AgendamentoLogProcessoModel.cs
using Oi.Juridico.Contextos.V2.AgendamentoLogProcessoContext.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Oi.Juridico.WebApi.V2.Areas.Relatorios.Models
{
    public class AgendamentoLogProcessoModel
    { 
        public decimal Id { get; set; }
        public DateTime DataIni { get; set; }
        public DateTime DataFim { get; set; }
        public DateTime DataDeAgendamento { get; set; }
        public DateTime DataDeCadastro { get; set; }
        public string UsuarioId { get; set; } = "";
        public string Operacao { get; set; } = "";
        public int TipoLog { get; set; }
        public int Status { get; set; }
        public string NomeDoArquivo { get; set; } = "";
        public string MensagemErro { get; set; } = "";
        public DateTime? DataIniExecucao { get;  set; }
        public DateTime? DataFimExecucao { get;  set; } 
        public string ArquivoBase64 { get; set; }

        public AgendamentoLogProcessoModel() {
            DataDeAgendamento = DateTime.Now;
            DataDeCadastro = DateTime.Now;
            Status = 1;
        } 
        public AgendLogProcesso retornaEntity(string userName)
        {
            return new AgendLogProcesso()
            {
                Id = Id,
                DatLogIni =  DataIni,
                DatLogFim =  DataFim,
                DatAgendamento = DataDeAgendamento,
                DatCadastro = DataDeCadastro,
                UsrCodUsuario = userName,
                CodOperacao = Operacao,
                CodTipoLog =  TipoLog,
                Status =  Status,
                NomArquivo =  NomeDoArquivo,
                MsgErro =  MensagemErro,
                DatExecucaoIni = DataIniExecucao,
                DatExecucaoFim = DataFimExecucao
        };
            
        }

    }
}
